<?xml version="1.0"?>
<!--XML-->
<launch>
    <!-- SETUP ARGUMENTS -->
    <!-- enable/disable running rviz in launcher -->
    <arg name="rviz" default="true" />
    <arg name="titania_ns" default="i3dr_titania" />

    <!-- Reduce point cloud size with VoxelGrid -->
    <!-- <node name="reduce_pcl" pkg="i3dr_detect_pcl_primitives" type="reduce_pcl" output="screen" >
        <param name="grid_size" type="double" value="0.04" />
        <remap from="/points2" to="$(arg titania_ns)/points2" /> 
        <remap from="/points2_reduced" to="$(arg titania_ns)/points2_reduced" />
    </node> -->

    <!-- Detect primitive in point cloud -->
    <node name="detect_pcl_primitives" pkg="i3dr_detect_pcl_primitives" type="detect_pcl_primitives" output="screen" >
        <remap from="/points2" to="$(arg titania_ns)/points2" />
    </node>

    <!-- DISPLAY -->
    <!-- run rviz with config file -->
    <node if="$(arg rviz)" pkg="rviz" type="rviz" name="rivz" 
        args="-d $(find i3dr_detect_pcl_primitives)/rviz/titania.rviz"/>

</launch>